lib/puppet/provider/exec/powershell.rb

```diff
@@ -68,7 +68,13 @@ def run(command, check = false)
         return super("cmd.exe /c \"\"#{native_path(command(:powershell))}\" #{legacy_args} -Command - < \"#{native_path}\"\"", check)
       end
     else
-      result = ps_manager.execute(command)
+      if working_dir = resource[:cwd]
+        self.fail "Working directory '#{working_dir}' does not exist" unless File.directory?(working_dir)
```

Comment:
I was trying to understand mostly if this is a good thing or not. Is `{working_dir}` always and only set explicitly by the user?